rootfs/api/models/release.py

```diff
@@ -218,6 +219,9 @@ def rollback(self, user, version=None):
                 raise DeisException('Cannot roll back to initial release.')
 
             prev = self.app.release_set.get(version=version)
+            if prev.failed:
+                raise DeisException('Cannot roll back to failed release.')
```

Comment:
we calculate version few line above if not passed.so even if we don't pass version if the previous release is failed it won't rollback.